### PR TITLE
[com_content] - Add tags filtering based on language filter

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #19387 .

### Summary of Changes
- unpublished tags are now not returned
- tags are returned based on Language Filter option setting

### Testing Instructions
- create some tags in different languages and set some to unpublished status
- create a Category List menu item and  set Filter Field to Tags

### Expected result
The category list tag filter works as expected

### Actual result
- tags are showed even if they are unpublished
- tags are showed in all languages always

### CODE REVIEW

**Comment by csbalazs** [comment]
It works!

**Comment by ghost** [comment]
@csbalazs please mark your Test as successfully:

- open [Issue Tracker](https://issues.joomla.org/tracker/joomla-cms/19509)
- Login with your github-Account
- Click on blue "Test this"-Button above Authors-Picture
- mark your Test as successfully
- hit "submit test result"

**Comment by csbalazs** [comment]
I have tested this item :white_check_mark: successfully on 16830cf841045d72ec0bd65d950f27407bd9afc5<br /><br />It works now! Thx!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19509">issues.joomla.org/tracker/joomla-cms/19509</a>.</sub>

**After the repairs:**
Site language: Hungarian, The tag filter only shows Hungarian tags.

Site language: English, The tag filter only shows English tags.

File: components/com_content/views/category/tmpl/default_articles.php [hunk 1]
```diff
@@ -9,13 +9,38 @@
...
-$n         = count($this->items);
-$listOrder = $this->escape($this->state->get('list.ordering'));
-$listDirn  = $this->escape($this->state->get('list.direction'));
+$params     = &$this->item->params;
+$n          = count($this->items);
+$listOrder  = $this->escape($this->state->get('list.ordering'));
+$listDirn   = $this->escape($this->state->get('list.direction'));
+$langFilter = false;
+
+//Tags filtering based on language filter
```
Comment (Quy) [inline components/com_content/views/category/tmpl/default_articles.php]: Add space after `//`

File: components/com_content/views/category/tmpl/default_articles.php [hunk 2]
```diff
@@ -9,13 +9,38 @@
...
+
+		case 'all' :
+			$langFilter = false;
+			break;
+
+		default :
+			$langFilter = ComponentHelper::getParams('com_tags')->get('tag_list_language_filter');
+	}
+}
+
```
Comment (Quy) [inline components/com_content/views/category/tmpl/default_articles.php]: Delete blank line.

File: components/com_content/views/category/tmpl/default_articles.php (hunk 1 above)
Comment (alikon) [reply components/com_content/views/category/tmpl/default_articles.php]: done

File: components/com_content/views/category/tmpl/default_articles.php (hunk 2 above)
Comment (alikon) [reply components/com_content/views/category/tmpl/default_articles.php]: done

**Comment by csbalazs** [comment]
I have tested this item :white_check_mark: successfully on c82a894776759ddf51cff78121fbce529e1f0f17<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19509">issues.joomla.org/tracker/joomla-cms/19509</a>.</sub>

File: components/com_content/views/category/tmpl/default_articles.php [hunk 3]
```diff
@@ -9,13 +9,37 @@
...
+		case 'current_language' :
+			$langFilter = JFactory::getApplication()->getLanguage()->getTag();
+			break;
+
+		case 'all' :
+			$langFilter = false;
+			break;
+
+		default :
+			$langFilter = ComponentHelper::getParams('com_tags')->get('tag_list_language_filter');
```
Comment (Quy) [inline components/com_content/views/category/tmpl/default_articles.php]: Change to `$langFilter = $tagfilter;`

Comment (alikon) [reply components/com_content/views/category/tmpl/default_articles.php]: yes

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 3cc5970e91ecee4a0137c0aeec22e10895f7eb9e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19509">issues.joomla.org/tracker/joomla-cms/19509</a>.</sub>

**Comment by csbalazs** [comment]
I have tested this item :white_check_mark: successfully on 3cc5970e91ecee4a0137c0aeec22e10895f7eb9e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19509">issues.joomla.org/tracker/joomla-cms/19509</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by florahwilliams** [comment]
English literature assignment writing services are essential for English literature coursework writing service students and [English Literature Writing Services](https://topessaywriting.com/english-literature-writing-services/ "English Literature Writing Services") seekers. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19509">issues.joomla.org/tracker/joomla-cms/19509</a>.</sub>

**Pull request merged**